src/extension.ts

```diff
@@ -110,4 +114,91 @@ export const activate = () => {
         if (!getExtensionSetting('runOnSave') || reason === vscode.TextDocumentSaveReason.AfterDelay) return
         waitUntil(performFixes())
     })
+
+    vscode.workspace.onDidChangeTextDocument(
+        ({ contentChanges, document }) => {
+            if (!getExtensionSetting("insertMissingCommaOnEnter")) {
+                return;
+            }
+
+            if (
+                !vscode.languages.match(['json', 'jsonc'], document)
+            ) {
+                return;
+            }
+
+            if (contentChanges.length === 0) {
+                return;
+            }
+
+            contentChanges = [...contentChanges].sort((a, b) => a.range.start.compareTo(b.range.start))
+
+            const editor = vscode.window.activeTextEditor;
+
+            if (
+                document.uri !== editor?.document.uri ||
+                ["output"].includes(editor.document.uri.scheme)
+            ) {
+                return;
+            }
+
+            if (
+                vscode.workspace.fs.isWritableFileSystem(
+                    document.uri.scheme
+                ) === false
+            ) {
+                return;
+            }
+
+            void editor.edit((edit) => {
+                for (const [i, content] of contentChanges.entries()) {
+                    if (
+                        !content.text.startsWith("\n") &&
+                        !content.text.startsWith("\r\n")
+                    ) {
+                        continue;
+                    }
+
+                    const prevLinePosition = content.range.start;
+
+                    const prevLine = document.lineAt(prevLinePosition.line + i);
+                    const prevLineText = prevLine.text;
+
+                    const prevLineLastChar = prevLineText.at(-1);
+
+                    if (!prevLineLastChar) {
+                        continue;
+                    }
+
+                    if (document.languageId === 'jsonc') {
```

Comment:
I think it would be better to remove this check, even if user still uses comments inside json (not jsonc) we still have to do comma placing at the right position.